app/main.py

```diff
@@ -1,34 +1,82 @@
+from typing import Tuple, List
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: Tuple[int],
+            end: Tuple[int],
+            is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = [
+            Deck(row=row, column=column)
+            for row in range(start[0], end[0] + 1)
+            for column in range(start[1], end[1] + 1)
+        ]
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if (deck.row == row) and (deck.column == column):
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row=row, column=column)
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if deck is not None:
+            deck.is_alive = False
+            self.decks.remove(deck)
```

Comment:
Use 'if deck' instead of 'if deck is not None' for a more concise and Pythonic approach